jenkins/version-def.sh

```diff
@@ -56,15 +56,28 @@ function set_env_var_SPARK_SHIM_VERSIONS_ARR() {
     SPARK_SHIM_VERSIONS_STR=$(echo $SPARK_SHIM_VERSIONS_STR)
     IFS=", " <<< $SPARK_SHIM_VERSIONS_STR read -r -a SPARK_SHIM_VERSIONS_ARR
 }
-# Psnapshots: snapshots + noSnapshots
-set_env_var_SPARK_SHIM_VERSIONS_ARR -Psnapshots
-SPARK_SHIM_VERSIONS_SNAPSHOTS=("${SPARK_SHIM_VERSIONS_ARR[@]}")
-# PnoSnapshots: noSnapshots only
-set_env_var_SPARK_SHIM_VERSIONS_ARR -PnoSnapshots
-SPARK_SHIM_VERSIONS_NOSNAPSHOTS=("${SPARK_SHIM_VERSIONS_ARR[@]}")
-# PsnapshotOnly : snapshots only
-set_env_var_SPARK_SHIM_VERSIONS_ARR -PsnapshotOnly
-SPARK_SHIM_VERSIONS_SNAPSHOTS_ONLY=("${SPARK_SHIM_VERSIONS_ARR[@]}")
+
+if [[ $SCALA_BINARY_VER == "2.13" ]]; then
+    # Psnapshots: snapshots + noSnapshots
+    set_env_var_SPARK_SHIM_VERSIONS_ARR -PsnapshotsScala213
+    SPARK_SHIM_VERSIONS_SNAPSHOTS=("${SPARK_SHIM_VERSIONS_ARR[@]}")
+    # PnoSnapshots: noSnapshots only
+    set_env_var_SPARK_SHIM_VERSIONS_ARR -PnoSnapshotsScala213
+    SPARK_SHIM_VERSIONS_NOSNAPSHOTS=("${SPARK_SHIM_VERSIONS_ARR[@]}")
+    # PsnapshotOnly : snapshots only
+    set_env_var_SPARK_SHIM_VERSIONS_ARR -PsnapshotScala213Only
+    SPARK_SHIM_VERSIONS_SNAPSHOTS_ONLY=("${SPARK_SHIM_VERSIONS_ARR[@]}")
+else
+    # Psnapshots: snapshots + noSnapshots
+    set_env_var_SPARK_SHIM_VERSIONS_ARR -Psnapshots
+    SPARK_SHIM_VERSIONS_SNAPSHOTS=("${SPARK_SHIM_VERSIONS_ARR[@]}")
+    # PnoSnapshots: noSnapshots only
+    set_env_var_SPARK_SHIM_VERSIONS_ARR -PnoSnapshots
+    SPARK_SHIM_VERSIONS_NOSNAPSHOTS=("${SPARK_SHIM_VERSIONS_ARR[@]}")
+    # PsnapshotOnly : snapshots only
+    set_env_var_SPARK_SHIM_VERSIONS_ARR -PsnapshotOnly
+    SPARK_SHIM_VERSIONS_SNAPSHOTS_ONLY=("${SPARK_SHIM_VERSIONS_ARR[@]}")
+fi
```

Comment:
@NVnavkumar  Can you also help to add `premergexxxScala213` similar to above changes for the premerge CI to run UTs? Thanks!

https://github.com/NVIDIA/spark-rapids/blob/branch-23.12/jenkins/version-def.sh#L96-L102

Reply:
Build against scala 2.13 failed as below with latest PR source changes:

`bash build/buildall -p=330 --scala213` or

```
bash build/make-scala-version-build-files.sh
cd scala2.13
mvn clean verify -Dbuildver=330

```

----------------------------------------------------------------------------------------------

```
[INFO] --- scala-maven-plugin:4.3.0:compile (scala-compile-first) @ rapids-4-spark-sql_2.13 ---
[INFO] Using incremental compilation using Mixed compile order
[INFO] Compiler bridge file: /home/timl/work.d/spark-rapids/scala2.13/target/spark330/.sbt/1.0/zinc/org.scala-sbt/org.scala-sbt-compiler-bridge_2.13-1.3.1-bin_2.13.8__52.0-1.3.1_20191012T045515.jar
[INFO] Compiling 419 Scala sources and 60 Java sources to /home/timl/work.d/spark-rapids/scala2.13/sql-plugin/target/spark330/classes ...
[ERROR] ## Exception when compiling 479 sources to /home/timl/work.d/spark-rapids/scala2.13/sql-plugin/target/spark330/classes
java.lang.IllegalStateException: don't know how to handle an output of /home/timl/work.d/spark-rapids/scala2.13/sql-plugin/target/spark330/classes [class scala.reflect.io.PlainFile]
scala.tools.nsc.backend.jvm.ClassfileWriters$FileWriter$.apply(ClassfileWriters.scala:185)
scala.tools.nsc.backend.jvm.ClassfileWriters$ClassfileWriter$.apply(ClassfileWriters.scala:78)
scala.tools.nsc.backend.jvm.PostProcessor.initialize(PostProcessor.scala:57)
scala.tools.nsc.backend.jvm.GenBCode$BCodePhase.initialize(GenBCode.scala:96)
scala.tools.nsc.backend.jvm.GenBCode$BCodePhase.$anonfun$run$1(GenBCode.scala:79)
scala.tools.nsc.backend.jvm.GenBCode$BCodePhase.run(GenBCode.scala:78)
scala.tools.nsc.Global$Run.compileUnitsInternal(Global.scala:1524)
scala.tools.nsc.Global$Run.compileUnits(Global.scala:1508)
scala.tools.nsc.Global$Run.compileSources(Global.scala:1500)
scala.tools.nsc.Global$Run.compile(Global.scala:1634)
xsbt.CachedCompiler0.run(CompilerInterface.scala:153)
xsbt.CachedCompiler0.run(CompilerInterface.scala:125)
xsbt.CompilerInterface.run(CompilerInterface.scala:39)
sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
java.lang.reflect.Method.invoke(Method.java:498)
sbt.internal.inc.AnalyzingCompiler.call(AnalyzingCompiler.scala:248)
sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:122)
sbt.internal.inc.AnalyzingCompiler.compile(AnalyzingCompiler.scala:95)
sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$4(MixedAnalyzingCompiler.scala:91)
scala.runtime.java8.JFunction0$mcV$sp.apply(JFunction0$mcV$sp.java:23)
sbt.internal.inc.MixedAnalyzingCompiler.timed(MixedAnalyzingCompiler.scala:186)
sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$3(MixedAnalyzingCompiler.scala:82)
sbt.internal.inc.MixedAnalyzingCompiler.$anonfun$compile$3$adapted(MixedAnalyzingCompiler.scala:77)
sbt.internal.inc.JarUtils$.withPreviousJar(JarUtils.scala:215)
sbt.internal.inc.MixedAnalyzingCompiler.compileScala$1(MixedAnalyzingCompiler.scala:77)
sbt.internal.inc.MixedAnalyzingCompiler.compile(MixedAnalyzingCompiler.scala:146)
sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1(IncrementalCompilerImpl.scala:343)
sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileInternal$1$adapted(IncrementalCompilerImpl.scala:343)
sbt.internal.inc.Incremental$.doCompile(Incremental.scala:120)
sbt.internal.inc.Incremental$.$anonfun$compile$4(Incremental.scala:100)
sbt.internal.inc.IncrementalCommon.recompileClasses(IncrementalCommon.scala:180)
sbt.internal.inc.IncrementalCommon.cycle(IncrementalCommon.scala:98)
sbt.internal.inc.Incremental$.$anonfun$compile$3(Incremental.scala:102)
sbt.internal.inc.Incremental$.manageClassfiles(Incremental.scala:155)
sbt.internal.inc.Incremental$.compile(Incremental.scala:92)
sbt.internal.inc.IncrementalCompile$.apply(Compile.scala:75)
sbt.internal.inc.IncrementalCompilerImpl.compileInternal(IncrementalCompilerImpl.scala:348)
sbt.internal.inc.IncrementalCompilerImpl.$anonfun$compileIncrementally$1(IncrementalCompilerImpl.scala:301)
sbt.internal.inc.IncrementalCompilerImpl.handleCompilationError(IncrementalCompilerImpl.scala:168)
sbt.internal.inc.IncrementalCompilerImpl.compileIncrementally(IncrementalCompilerImpl.scala:248)
sbt.internal.inc.IncrementalCompilerImpl.compile(IncrementalCompilerImpl.scala:74)
sbt_inc.SbtIncrementalCompiler.compile(SbtIncrementalCompiler.java:173)
scala_maven.ScalaCompilerSupport.incrementalCompile(ScalaCompilerSupport.java:297)
scala_maven.ScalaCompilerSupport.compile(ScalaCompilerSupport.java:109)
scala_maven.ScalaCompilerSupport.doExecute(ScalaCompilerSupport.java:91)
scala_maven.ScalaMojoSupport.execute(ScalaMojoSupport.java:554)
org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:137)
org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:210)
org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:156)
org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:148)
org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:117)
org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:81)
org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:56)
org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:128)
org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:305)
org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:192)
org.apache.maven.DefaultMaven.execute(DefaultMaven.java:105)
org.apache.maven.cli.MavenCli.execute(MavenCli.java:956)
org.apache.maven.cli.MavenCli.doMain(MavenCli.java:288)
org.apache.maven.cli.MavenCli.main(MavenCli.java:192)
sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
java.lang.reflect.Method.invoke(Method.java:498)
org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
           
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary for RAPIDS Accelerator for Apache Spark Root Project 23.12.0-SNAPSHOT:
[INFO] 
[INFO] RAPIDS Accelerator for Apache Spark Root Project ... SUCCESS [ 12.816 s]
[INFO] rapids-4-spark-sql-plugin-api_2.13 ................. SUCCESS [ 12.579 s]
[INFO] RAPIDS Accelerator for Apache Spark SQL Plugin ..... FAILURE [ 42.651 s]
[INFO] RAPIDS Accelerator for Apache Spark Shuffle Plugin . SKIPPED
[INFO] RAPIDS Accelerator for Apache Spark Scala UDF Plugin SKIPPED
[INFO] RAPIDS Accelerator for Apache Spark Delta Lake 2.1.x Support SKIPPED
[INFO] RAPIDS Accelerator for Apache Spark Delta Lake 2.2.x Support SKIPPED
[INFO] RAPIDS Accelerator for Apache Spark Aggregator ..... SKIPPED
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:08 min
[INFO] Finished at: 2023-10-19T22:13:09+08:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal net.alchim31.maven:scala-maven-plugin:4.3.0:compile (scala-compile-first) on project rapids-4-spark-sql_2.13: Execution scala-compile-first of goal net.alchim31.maven:scala-maven-plugin:4.3.0:compile failed: don't know how to handle an output of /home/timl/work.d/spark-rapids/scala2.13/sql-plugin/target/spark330/classes [class scala.reflect.io.PlainFile] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
[ERROR] 
[ERROR] After correcting the problems, you can resume the build with the command
[ERROR]   mvn <goals> -rf :rapids-4-spark-sql_2.13
xargs: bash: exited with status 255; aborting

real    1m11.774s
user    3m56.288s
sys     0m3.080s
```

Reply:
Hmm, I don't know how to replicate this. What JDK version are you compiling with?

Reply:
> Hmm, I don't know how to replicate this. What JDK version are you compiling with?

My local java version is java8 1.8.0.
```
~$ java -version
openjdk version "1.8.0_352"
OpenJDK Runtime Environment (build 1.8.0_352-8u352-ga-1~18.04-b08)
OpenJDK 64-Bit Server VM (build 25.352-b08, mixed mode)
```
I could always get the failure on my local host when building scala 2.13, but building scala 2.12 was good.
Let me also check in the pre-merge docker container to see if it is reproducible.

Reply:
I cannot reproduce either, may check your local permissions of related folders
```
/home/timl/work.d/spark-rapids/scala2.13/target/spark330/
```

Reply:
> I cannot reproduce either, may check your local permissions of related folders
> 
> ```
> /home/timl/work.d/spark-rapids/scala2.13/target/spark330/
> ```

Yes, should be local environment issue.
Build against 2.13 can success in docker container, too.

Please ignore it. Let me double check my local env.